include/knowhere/index/index_node.h

```diff
@@ -437,6 +442,10 @@ class IndexNode : public Object {
 //   with quantization, override `raw_distance`.
 // Internally, this structure uses the same priority queue class, but may multiply all
 //   incoming distances to (-1) value in order to turn max priority queue into a min one.
+// If use_knowhere_search_pool is True (the default), the iterator->Next() will be scheduled by the
+//   knowhere_search_thread_pool.
+//   If False, will Not involve thread scheduling internally, so please take caution.
+template <bool use_knowhere_search_pool = true>
```

Comment:
What are the justifications for adding `use_knowhere_search_pool` as a template parameter instead of making a regular `IndexIterator::use_knowhere_search_pool` field? I'm not sure I follow

Reply:
Just to reduce the judgment of `use_knowhere_search_pool` in each `iterator->next()` call.

Reply:
I've re-read the code again and I don't believe I see a reason for making `use_knowhere_search_pool` a template parameter. Basically, it leads to increase of the size of the compiled code, also this parameter is not generic enough (like `T` in `std::vector<T>`) and it does not have a critical-performance impact. Please consider demoting `use_knowhere_search_pool` to a variable. Or I can accept the code as is.

Reply:
Thanks for your suggestion~ I will use `use_knowhere_search_pool` as an iterator variable rather than a template param.